### PR TITLE
Check template for mesh

### PR DESCRIPTION
**Pull request by itziakos**
This PR add a set template objects to check conformance for Mesh containers.

in more details:

- Separate template classes to check functionality of point, edge, face and cell operations.
- Separate template class to check Mesh attributes functionality
- Updated Mesh and H5Mesh tests to use the templates.
- Fixed bug where the number of connected points was not updated.
- Updated doc-strings to be consistent with the currently expected behaviour.
- Minor cleanups here and there.

### CODE REVIEW

**Comment by itziakos** [comment]
@roigcarlo and @nathanfranklin this PR is now ready for review. Please note that we improved the code coverage by ~ 4%.

File: simphony/cuds/mesh.py [hunk 1]
```diff
@@ -495,23 +478,12 @@ def update_point(self, point):
         KeyError
```
Comment (nathanfranklin) [inline simphony/cuds/mesh.py]: ValueError

Comment (itziakos) [inline simphony/cuds/mesh.py]: Done

**Comment by nathanfranklin** [comment]
:+1:

**Comment by itziakos** [comment]
@roigcarlo and @nathanfranklin, I have updated the PR to include a set of tests to target containers that do not support all the cuba keys.

**Comment by nathanfranklin** [comment]
:+1:

**Pull request merged**